web/src/containers/ApdPreviousActivityTableMMIS.js

```diff
@@ -0,0 +1,144 @@
+        <tbody>
+          {years.map(year => {
+            const total = rollup(previousActivityExpenses[year]);
+
+            return (
+              <tr key={year} className="align-middle">
+                <th>{t('ffy', { year })}</th>
+                <td>
+                  <DollarInput
+                    name={`approved-federal-mmis-${year}`}
+                    label={`approved federal share for mmis, FFY ${year}`}
+                    hideLabel
+                    wrapperClass="m0"
+                    className="m0 input input-condensed mono right-align"
+                    value={previousActivityExpenses[year].mmis.federalApproved}
+                    onChange={handleChange(year, 'mmis', 'federalApproved')}
+                  />
+                </td>
+                <td>
+                  <DollarInput
+                    name={`actual-federal-mmis-${year}`}
+                    label={`actual federal share for mmis, FFY ${year}`}
+                    hideLabel
+                    wrapperClass="m0"
+                    className="m0 input input-condensed mono right-align"
+                    value={previousActivityExpenses[year].mmis.federalActual}
+                    onChange={handleChange(year, 'mmis', 'federalActual')}
+                  />
+                </td>
+                <td>
+                  <DollarInput
+                    name={`approved-state-mmis-${year}`}
+                    label={`approved state share for mmis, FFY ${year}`}
+                    hideLabel
+                    wrapperClass="m0"
+                    className="m0 input input-condensed mono right-align"
+                    value={previousActivityExpenses[year].mmis.stateApproved}
+                    onChange={handleChange(year, 'mmis', 'stateApproved')}
+                  />
+                </td>
+                <td>
+                  <DollarInput
+                    name={`actual-state-mmis-${year}`}
+                    label={`actual state share for mmis, FFY ${year}`}
+                    hideLabel
+                    wrapperClass="m0"
+                    className="m0 input input-condensed mono right-align"
+                    value={previousActivityExpenses[year].mmis.stateActual}
+                    onChange={handleChange(year, 'mmis', 'stateActual')}
+                  />
+                </td>
+                <td>{formatMoney(total.approved)}</td>
+                <td>{formatMoney(total.actual)}</td>
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+
+ApdPreviousActivityTable.propTypes = {
+  previousActivityExpenses: PropTypes.object.isRequired,
+  updateApd: PropTypes.func.isRequired
+};
+
+const mapStateToProps = ({ apd: { data: { previousActivityExpenses } } }) => ({
+  previousActivityExpenses
+});
+
+const mapDispatchToProps = { updateApd };
+
+export default connect(mapStateToProps, mapDispatchToProps)(
+  ApdPreviousActivityTable
+);
```

Comment:
This should probably be abstracted in such a way we can use one component for both tables (I don't think it should be TOO hard), but for the sake of just getting it knocked out, I duplicated it and removed the unnecessary bits.

Reply:
yeah, makes sense, can you make an issue to revisit / potentially consolidate?